api/V1/Recording.js

```diff
@@ -84,11 +84,41 @@ module.exports = (app, baseUrl) => {
    * @apiSuccess {Number} recordingId ID of the recording.
    * @apiuse V1ResponseError
    */
+
   app.post(
-    apiUrl + "/:devicename",
+    apiUrl + "/device/:devicename/group/:groupname?",
     [
       auth.authenticateUser,
-      middleware.getDeviceByName(param),
+      middleware.setGroupName(param),
+      middleware.getDevice(param),
+      auth.userCanAccessDevices
+    ],
+    middleware.requestWrapper(recordingUtil.makeUploadHandler())
+  );
+
+  /**
+   * @api {post} /api/v1/recordings/device/:deviceId Add a new recording on behalf of device
+   * @apiName PostRecordingOnBehalf
+   * @apiGroup Recordings
+   * @apiDescription Called by a user to upload raw thermal video on behalf of a device.
+   * The user must have permission to view videos from the device or the call will return an
+   * error.  It currently supports raw thermal video but will eventually support all
```

Comment:
Please remove the bit about this only working for thermal video. Same for the other POST API that takes group name